### PR TITLE
Support for creating and deleting (manual) snapshots

### PR DESCRIPTION
**Pull request by tenyo**

### CODE REVIEW

File: actions/snapshots.go [hunk 1]
```diff
@@ -12,6 +12,74 @@ import (
...
 
+// SnapshotsPost creates a manual snapshot for a given database instance or cluster
+func SnapshotsPost(c buffalo.Context) error {
+	req := SnapshotCreateRequest{}
+	if err := c.Bind(&req); err != nil {
+		log.Println(err)
+		return c.Error(400, err)
+	}
+
+	if req.SnapshotIdentifier == nil {
```
Comment (fishnix) [inline actions/snapshots.go]: you also want the string to be not empty i think.  i go back and forth on using pointers of concrete types in request/response structs.  its useful in a lot of places to just check `!= nil`, but this is an example where more is required.

**Review by fishnix — COMMENTED** [review]
I wonder if we could improve those handlers using the orchestrator pattern - have a separate function of `clusterSnapshotCreate` and `instanceSnapshotCreate` that does your special error parsing and returns the clusterSnapshot and err.  Then i think your handler could look more like:

```golang
    output := struct {
		DBClusterSnapshot *rds.DBClusterSnapshot `json:"DBClusterSnapshot,omitempty"`
		DBSnapshot        *rds.DBSnapshot        `json:"DBSnapshot,omitempty"`
	}{}

    clusterSnapshot, err := o.clusterSnapshotCreate(db, snap)
    if err....
    output.DBClusterSnapshot = clusterSnapshot

    if clusterSnapshot == nil {
        instanceSnapshot, err := instanceSnapshotCreate(db, snap)
        if err...
        output.DBSnapshot = instanceSnapshot
    }

    return c.Render(200, r.JSON(output))    
```

and in the orchestrators you could return to avoid the nested conditionals.

**Comment by tenyo** [comment]
> I wonder if we could improve those handlers using the orchestrator pattern - have a separate function of `clusterSnapshotCreate` and `instanceSnapshotCreate` that does your special error parsing and returns the clusterSnapshot and err. Then i think your handler could look more like:
> 
> ```go
>     output := struct {
> 		DBClusterSnapshot *rds.DBClusterSnapshot `json:"DBClusterSnapshot,omitempty"`
> 		DBSnapshot        *rds.DBSnapshot        `json:"DBSnapshot,omitempty"`
> 	}{}
> 
>     clusterSnapshot, err := o.clusterSnapshotCreate(db, snap)
>     if err....
>     output.DBClusterSnapshot = clusterSnapshot
> 
>     if clusterSnapshot == nil {
>         instanceSnapshot, err := instanceSnapshotCreate(db, snap)
>         if err...
>         output.DBSnapshot = instanceSnapshot
>     }
> 
>     return c.Render(200, r.JSON(output))    
> ```
> 
> and in the orchestrators you could return to avoid the nested conditionals.

That seems like it will make the code harder to read and more complicated. Maybe if it was more complex orchestration it would make sense.

**Review by fishnix — APPROVED** [review]
lgtm! 👍

**Pull request merged**